### PR TITLE
feat: Migrate from legacy /vuln endpoint to /test

### PR DESCRIPTION
**Pull request by robcresswell**
This PR migrates the vulncost plugin to use the `/test` endpoint instead of `/vuln`, which is being removed soon. Since `/test` does not support resolving `latest` as a version (which is an NPM specific concept), we shift the version resolution to the plugin

### CODE REVIEW

File: src/getImports/testVuln.js [hunk 1]
```diff
@@ -4,30 +4,27 @@ import logger from '../logger';
...
 function testNoAuth(key) {
-  return axios
-    .get(`https://snyk.io/test/npm/${key}?${utm}&type=json`)
-    .then(({ data }) => {
-      if (typeof data === 'string') {
-        // bug on snyk's side, returning a string for 404
-        logger.log('bad return on ' + key);
-        throw new Error('bad return from snyk api (unauthed)');
-      }
+  return axios.get(`${API_ROOT}${key}?${utm}&type=json`).then(({ data }) => {
```
Comment (Ryuno-Ki) [inline src/getImports/testVuln.js]: Might make sense to turn this into an async / await function as well?

Comment (robcresswell) [reply src/getImports/testVuln.js]: It makes sense for it to happen at some point, but I'm trying to keep the patch scope as minimal as possible. The risk involved here is already fairly high.

Comment (Ryuno-Ki) [reply src/getImports/testVuln.js]: Understood.
Perhaps worth to fill another PR once this one landed.

**Review by bmvermeer — APPROVED** [review]
LGTM

**Pull request merged**